app/services/journeyAnswers/IncomeAnswersService.scala

```diff
@@ -18,30 +18,67 @@ package services.journeyAnswers
+import repositories.JourneyAnswersRepository
+import services.mapDownstreamErrors
+import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.ExecutionContext
 
 trait IncomeAnswersService {
-  def saveAnswers(businessId: BusinessId, taxYear: TaxYear, mtditid: Mtditid, answers: IncomeJourneyAnswers): ApiResultT[Unit]
+  def saveAnswers(ctx: JourneyContextWithNino, answers: IncomeJourneyAnswers)(implicit hc: HeaderCarrier): ApiResultT[Unit]
 }
 
 @Singleton
-class IncomeAnswersServiceImpl @Inject() (repository: MongoJourneyAnswersRepository)(implicit ec: ExecutionContext) extends IncomeAnswersService {
-
-  def saveAnswers(businessId: BusinessId, taxYear: TaxYear, mtditid: Mtditid, answers: IncomeJourneyAnswers): ApiResultT[Unit] =
-    EitherT
-      .right[ServiceError](
-        repository.upsertData(JourneyContext(taxYear, businessId, mtditid, Income), Json.toJson(answers))
-      )
-      .void
+class IncomeAnswersServiceImpl @Inject() (repository: JourneyAnswersRepository, connector: SelfEmploymentBusinessConnector)(implicit
+    ec: ExecutionContext)
+    extends IncomeAnswersService {
+
+  def saveAnswers(ctx: JourneyContextWithNino, answers: IncomeJourneyAnswers)(implicit hc: HeaderCarrier): ApiResultT[Unit] = {
+    import ctx._
+    val createIncome = IncomesType(answers.turnoverIncomeAmount.some, answers.nonTurnoverIncomeAmount)
+    val createBody   = CreateSEPeriodSummaryRequestBody(startDate(taxYear), endDate(taxYear), Some(FinancialsType(createIncome.some, None)))
+    val createData   = CreateSEPeriodSummaryRequestData(taxYear, businessId, nino, createBody)
+
+    val amendIncome = Incomes(answers.turnoverIncomeAmount.some, answers.nonTurnoverIncomeAmount, None)
+    val amendBody   = AmendSEPeriodSummaryRequestBody(amendIncome.some, None)
+    val amendData   = AmendSEPeriodSummaryRequestData(taxYear, nino, businessId, amendBody)
+
+    val upsertBody = CreateAmendSEAnnualSubmissionRequestBody(
+      Some(AnnualAdjustments.empty.copy(includedNonTaxableProfits = answers.notTaxableAmount, outstandingBusinessIncome = answers.otherIncomeAmount)),
+      Some(AnnualAllowances.empty.copy(tradingIncomeAllowance = answers.tradingAllowanceAmount)),
+      None
+    )
+    val upsertData = CreateAmendSEAnnualSubmissionRequestData(taxYear, nino, businessId, upsertBody)
+
+    val result = for {
+      _        <- EitherT.right[DownstreamError](repository.upsertData(JourneyContext(taxYear, businessId, mtditid, journey), Json.toJson(answers)))
+      response <- EitherT(connector.listSEPeriodSummary(ctx))
+      _ <-
+        if (submissionExists(response)) EitherT(connector.amendSEPeriodSummary(amendData)) else EitherT(connector.createSEPeriodSummary(createData))
+      _ <- EitherT(connector.createAmendSEAnnualSubmission(upsertData))
+    } yield ()
+
+    result.leftMap(mapDownstreamErrors)
+  }
+
+  private def submissionExists(response: ListSEPeriodSummariesResponse) =
```

Comment:
I think it would be safer to just use the Api1965Response case class directly instead of low level json?

Reply:
Good point! Much simpler too.